contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -536,7 +540,7 @@ func (s *SchedulerServer) getDriver() (driver bindings.SchedulerDriver) {
 }
 
 func (s *SchedulerServer) Run(hks hyperkube.Interface, _ []string) error {
-	if n := len(s.mesosRoles); n == 0 || n > 2 || (n == 2 && s.mesosRoles[0] != "*" && s.mesosRoles[1] != "*") {
+	if n := len(s.frameworkRoles); n == 0 || n > 2 || (n == 2 && s.frameworkRoles[0] != "*" && s.frameworkRoles[1] != "*") {
```

Comment:
Use-case:
By default we set frameworkRoles to `*,slave_public` and defaultPodRoles also to `*,slave_public`. The user can customize the frameworkRoles, e.g. decide to allow only `*` roles, i.e. set frameworkRoles to `*` only. Then `defaultPodRoles` does not change.